packages/gatsby/src/redux/reducers/queries.ts

```diff
@@ -220,6 +220,14 @@ export function queriesReducer(
       state.dirtyQueriesListToEmitViaWebsocket = []
       return state
     }
+    case `MERGE_WORKER_QUERY_STATE`: {
+      assertSaneWorkerState(action.payload)
+
+      for (const workerState of action.payload) {
+        state = mergeWorkerDataDependencies(state, workerState)
+      }
+      return state
+    }
```

Comment:
TODO: here `action.payload` should contain `workerState` and not the array.

Reply:
Also need some tests for this action in https://github.com/gatsbyjs/gatsby/blob/master/packages/gatsby/src/redux/reducers/__tests__/queries.ts